### PR TITLE
test(client): Add test for mapped string with non-ascii characters on sqlserver

### LINKED ISSUE #22098: The column does not exist in the current database.
### Bug description
When executing find on the schema below it returns an error
The column `TESTE_NÃMERICO` does not exist in the current database.

In versions <= 5.3.0 it works

### How to reproduce
Clone `https://github.com/baglianiinovaetech/ErrorMapPrisma` and execute start

Or

Used example "https://www.prisma.io/docs/getting-started/quickstart" with the schema below

Create file script.ts

```ts
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

async function main() {
  await prisma.test.findFirst();
}

main()
  .then(async () => {
    await prisma.$disconnect();
  })
  .catch(async (e) => {
    console.error(e);
    await prisma.$disconnect();
    process.exit(1);
  });
```

run npx ts-node script.ts

The column `TESTE_NÃMERICO` does not exist in the current database.

### Expected behavior

_No response_

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "sqlserver"
  url      = env("DATABASE_URL")
}

model test {
    id             Int    @default(autoincrement())
    TESTE_N_MERICO String @unique @map("TESTE_NÚMERICO")
}
```
```ts
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

async function main() {
  await prisma.test.findFirst({});
}

main()
  .then(async () => {
    await prisma.$disconnect();
  })
  .catch(async (e) => {
    console.error(e);
    await prisma.$disconnect();
    process.exit(1);
  });

```

### Environment & setup

- OS: Windows 10
- Database:  SQL Server
- Node.js version: v18.17.1
- Yarn version: v1.22.19

### Prisma Version

```
 "@prisma/client": "^5.6.0",
 "prisma": "^5.6.0",
```

### PR DESCRIPTION
**Pull request by Druue**
Issue is fixed per 5.10
closes https://github.com/prisma/prisma/issues/22098

### CODE REVIEW

**Review by miguelff — APPROVED** [review]
Added a small comment, but it's not blocking feedback.

File: packages/client/tests/functional/issues/22098-column_does_not_exist/test.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+declare let prisma: PrismaClient
+
+testMatrix.setupTestSuite(() => {
+  test('does not throw error', async () => {
+    const t = await prisma.test.findFirst()
+
+    expect(t).toBe(null)
```
Comment (miguelff) [inline packages/client/tests/functional/issues/22098-column_does_not_exist/test.ts:11]:
```suggestion
    expect(prisma.test.findFirst()).resolve.not.toThrow()
```

maybe?

Comment (miguelff) [reply packages/client/tests/functional/issues/22098-column_does_not_exist/test.ts:11]: https://jestjs.io/docs/expect#resolves

File: packages/client/tests/functional/issues/22098-column_does_not_exist/test.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+declare let prisma: PrismaClient
+
+testMatrix.setupTestSuite(() => {
+  test('does not throw error', () => {
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises, jest/valid-expect
+    expect(prisma.test.findFirst()).resolves.not.toThrow()
```
Comment (Druue) [inline packages/client/tests/functional/issues/22098-column_does_not_exist/test.ts:10]: Went with the following solution, unfortunately, eslint doesn't seem to be able to pick up what's happening with the promise

**Pull request merged**